### PR TITLE
MWPW-143426 fixing sidenav parents when implicitly declared

### PR DESCRIPTION
**Pull request by npeltier**
when parents are auto-generated, they don't work well (no value => no unfold, no filtering)

Resolves: [MWPW-143426](https://jira.corp.adobe.com/browse/MWPW-143426)

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.page/drafts/npeltier/catalog?martech=off
- After: https://mwpw-143426--cc--npeltier.hlx.page/drafts/npeltier/catalog?martech=off

### CODE REVIEW

File: test/blocks/sidenav/sidenav.test.html.js [hunk 1]
```diff
@@ -41,20 +47,31 @@ runTests(async () => {
...
     });
 
     it('does create nice reordered categories sidenav', async () => {
       // 16 items from html requirements, 2 parents from taxonomy,
       // and special-offer special case is 19
-      await testCategorySidenav('.reordered-categories', 19, 13);
+      await testCategorySidenav('.reordered-categories', 19, 6, 13);
+    });
+
+    it('does put implicit at last position', async () => {
```
Comment (yesil) [inline test/blocks/sidenav/sidenav.test.html.js:71]: >does put implicit at last position
hard to understand what it does at first, possible to provide a more functional title?

Comment (npeltier) [reply test/blocks/sidenav/sidenav.test.html.js]: updated! note this comes out of a conversation with Geri, but i'm not sure anymore we really want this. I had lot of trouble creating a real use case for this. This PR however fixes a few problems it had anyway

**Comment by sharg1** [comment]
@npeltier  - there is lint check failures.. could you check?

`'implicitParents' is assigned a value but never used.`

**Review by 3ch023 — APPROVED** [review]

**Comment by afmicka** [comment]
looks good on my end. Will label as verified.

**Pull request merged**